### DISCUSSION TITLE
Merged relational+boolean operators?

### DISCUSSION THREAD

**Discussion by Daynvheur** [dotnet/csharplang#4980] [Language Ideas] [answered]
Hi, when checking for out of bounds values, I often repeat the same var name over, and over.
Example: `if (j < dateStart || j > dateEnd) throw new NotImplementedException();`

I'd like to have this variable name `j` not repeated while still checked for both boundaries.

As I didn't found a way to achieve it in one operation, I propose the implementation of merged comparison operators (not every may have real use cases):
- `<||>` 'inf left or sup right', ex: `j <||> (dateStart, dateEnd)`
- `<||=` 'inf left or equ right'
- `<||=>` 'inf left or sup or equ right'
- `=||>` 'equ left or sup right'
- `=||=` 'equ left or equ right', ex: `if (j =||= (dateStart, dateEnd)) ;` is similar with _switch expression_: `if (j switch { dateStart => true, dateEnd => true, _ => false }) ;`
- `=||=>` 'equ left or sup or equ right'
- `<=||>` 'inf or equ left or sup right'
- `<=||=` 'inf or equ left or equ right'
- `<=||=>` 'inf or equ left or sup or equ right'
(Same for `&&` operator instead of `||`, and optionally allowing `!=` comparisons.)

Would you find those usefull? Any thoughts about this?

**Comment by alrz** [comment]
See https://github.com/dotnet/csharplang/issues/4108

**Reply by Daynvheur** [reply]
Ternary comparison operators looks interesting (I've alreay read this proposal a while ago, couldn't find it when opening here), I would comment on there, as there is a slight difference in the treatment of `A<C>B` vs `A<||> (C,B)`.

**Comment by Richiban** [comment, marked as answer]
Since we have relational patterns now, you should be able to write this:

```cs
if (j is < dateStart or > dateEnd) throw new NotImplementedException();
```

```cs
if (j is x or y or z or < 1000) throw new NotImplementedException();
```

But unfortunately they don't allow comparisons with variables, only constants. I'd much rather we augmented this existing syntax to allow non-constant comparisons; I think that's a lot better than introducing new operators.

**Reply by Daynvheur** [reply]
Relational patterns are something I'd like to use more, if only it wasn't specifically for constants!
